guides/index-template.md

```diff
@@ -0,0 +1,304 @@
+LowLevelClusterNamespace cluster = client.Cluster; 
+```
+
+
+# Index Template API Actions
+
+
+## Create an Index Template
+You can create an index template to define default settings and mappings for indices of certain patterns. The following example creates an index template named `books` with default settings and mappings for indices of the `books-*` pattern:
+
+```csharp
+var createResponse = client.Indices.PutTemplateForAll<StringResponse>("books", PostData.String(@"{
+    ""index_patterns"": [""books-*""],
+    ""settings"": {
+        ""number_of_shards"": 3,
+        ""number_of_replicas"": 0
+    },
+    ""mappings"": {
+        ""properties"": {
+            ""title"": { ""type"": ""text"" },
+            ""author"": { ""type"": ""text"" },
+            ""published_on"": { ""type"": ""date"" },
+            ""pages"": { ""type"": ""integer"" }
+        }
+    }
+}")).Body;
+
+Console.WriteLine($"Create response: {createResponse}") // Create response: {"acknowledged":true}
+```
+
+Now, when you create an index that matches the `books-*` pattern, OpenSearch will automatically apply the template's settings and mappings to the index. Let's create an index named books-nonfiction and verify that its settings and mappings match those of the template:
+
+```csharp
+var createResponse = client.Indices.Create<StringResponse>("books-nonfiction", PostData.String(@"{
+    ""settings"": {
+        ""number_of_shards"": 3,
+        ""number_of_replicas"": 0
+    },
+    ""mappings"": {
+        ""properties"": {
+            ""title"": { ""type"": ""text"" },
+            ""author"": { ""type"": ""text"" },
+            ""published_on"": { ""type"": ""date"" },
+            ""pages"": { ""type"": ""integer"" }
+        }
+    }
+}")).Body;
+
+Console.WriteLine($"Create response: {createResponse}"); // Create response: {"acknowledged":true,"shards_acknowledged":true,"index":"books-nonfiction"}
+```
+
+
+## Multiple Index Templates
+If multiple index templates match the index's name, OpenSearch will apply the template with the highest `priority`. The following example creates two index templates named `books-*` and `books-fiction-*` with different settings:
+
+```csharp
+var createResponseOne = client.Indices.PutTemplateV2ForAll<StringResponse>("books", PostData.String(@"{
+    ""index_patterns"": [""books-*""],
+    // ""priority"": 1,
+    ""settings"": {
+        ""number_of_shards"": 3,
+        ""number_of_replicas"": 0
+    }
+}")).Body;
+
+Console.WriteLine($"Create response: {createResponseOne}"); // Create response: {"acknowledged":true}
+
+var createResponseTwo = client.Indices.PutTemplateV2ForAll<StringResponse>("books", PostData.String(@"{
+    ""index_patterns"": [""books-fiction-*""],
+    ""priority"": 2,
+    ""settings"": {
+        ""number_of_shards"": 3,
+        ""number_of_replicas"": 0
+    }
+}")).Body;
+
+Console.WriteLine($"Create response: {createResponseTwo}"); // Create response: {"acknowledged":true}
```

Comment:
The second template should be named `books-fiction`, and the settings of it should differ from the first
```suggestion
client.LowLevel.Indices.PutTemplateV2ForAll<VoidResponse>("books", PostData.Serializable(new
{
    index_patterns = new[] { "books-*" },
    priority = 0,
    template = new
    {
        settings = new
        {
            index = new
            {
                number_of_shards = 3,
                number_of_replicas = 0
            }
        }
    }
}));

client.LowLevel.Indices.PutTemplateV2ForAll<VoidResponse>("books-fiction", PostData.Serializable(new
{
    index_patterns = new[] { "books-fiction-*" },
    priority = 1,  // higher priority than the `books` template
    template = new
    {
        settings = new
        {
            index = new
            {
                number_of_shards = 1,
                number_of_replicas = 1
            }
        }
    }
}));
```